### PR TITLE
Makes xenos a subtype of human

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Your honor, I have never sprited a damn thing in my life.

Monkeys were made a subtype of Humans in https://github.com/tgstation/tgstation/pull/55614 - So it's important that Xenos do to.

Why?
Because:
1) It's the only thing preventing us from removing attack_paw and attack_alien
2) Humans, Monkeys and Xenos were all subtypes of /carbon. With Monkeys and Xenos gone, we can remove the subtype entirely, there's no reason for it to exist, it's not good, especially for maintaining.

Doing this ALSO means we can remove some vars from carbon for things like unique gibs, as they aren't needed anymore when species has vars to do the EXACT same things. Now that Xenos are a species, they can just use species' vars. Hear me?

Additional things this PR does:

- Removes attack_larva
- Removes attack_alien
- Refactors xeno queen pinpointer to be a subtype of agent pinpointer
- Refactors xeno hunter's tackling to use tackling component
- Refactors xeno HUDs to be the same as every other HUD.

## Why It's Good For The Game

It'll make xenos consistent with other mobs, and make it easier to maintain in the future.

## Changelog

:cl:
refactor: Xenomorphs have been made into a subtype of Humans, please report any bugs you see.
/:cl:

### CODE REVIEW

**Comment by nianjiilical** [comment]
please tell me this means shooting up the queen with 60u of bath salts is now a valid strategy

**Comment by JohnFulpWillard** [comment]
> please tell me this means shooting up the queen with 60u of bath salts is now a valid strategy

pretty sure it already is

**Comment by IndieanaJones** [comment]
This is a very scary PR but if you get it to work I'll be very happy

**Comment by Googles-Hands** [comment]
Looks neat, but is this freeze exempt or will this be closed soon?

**Comment by JohnFulpWillard** [comment]
> Looks neat, but is this freeze exempt or will this be closed soon?

The freeze doesn't stop refactors, only QoL/balance/content expansion.

**Comment by Mothblocks** [comment]
Monkeys were made human subtypes because basically every interaction we want humans to do, we want monkeys to do as well.

I'm not sure the same goes for xenos (who barely even have *hands*), and killing `/carbon`, while nice, isn't a convincing enough case to me to justify this change, especially when we do want some level of separation.

Can you point to real cases where the distinction is a problem, especially one where being a human species is a good idea?

**Comment by Mothblocks** [comment]
I'm also uncomfortable by the code--the idea behind species should be that we can easily apply them, not *require* the use of the species human subtypes, which only exist as presets.

**Comment by JohnFulpWillard** [comment]
> Can you point to real cases where the distinction is a problem, especially one where being a human species is a good idea?

attack_larva, attack_alien and attack_paw all only exist because of Xenos. These dumb procs should be removed (imo), and removing xeno's need to have them will let us remove 3 right there.

There's also vars on the carbon level, with xenos being the sole reason for it. On species, they're already used, because that's what most species do nowadays. But xenos are still stuck using carbon.

The species thing also isn't much of a problem, I took arm's (regrettably) idea of making them all subtypes, like Golems.
This is still a draft so I don't know what will be final, but I want to move out as much as I can, to the species, and have the least amount of things on the mob as I possibly can.

It will also give side effects like being able to make an admin button to turn someone into a xeno antag that would actually make them a xeno (like monkey fever), not having limbs just snowflake qdel when removed, and merge some emotes that for some reason did the exact same thing but one was for HUMAN/MONKEY and the other was for XENO. There's also some things that just do ishuman || isalien, when I don't think should be needed.

> I'm also uncomfortable by the code

I'm just trying my best here and seeing if I can do anything good.

File: code/__DEFINES/is_helpers.dm [hunk 1]
```diff
@@ -78,19 +78,19 @@ GLOBAL_LIST_INIT(turfs_without_ground, typecacheof(list(
 
 //more carbon mobs
 
-#define isalien(A) (istype(A, /mob/living/carbon/alien))
+#define isalien(A) (istype(A, /mob/living/carbon/human/species/alien))
```
Comment (Fikou) [inline code/__DEFINES/is_helpers.dm:81]: this no longer checks for larva

**Comment by Tennessee116** [comment]
I feel that this PR will be dragged down by code written to cancel out various unnecessary `/human` features.

File: code/__DEFINES/is_helpers.dm (hunk 1 above)
Comment (JohnFulpWillard) [reply code/__DEFINES/is_helpers.dm]: neither does your mom

**Comment by JohnFulpWillard** [comment]
species' attack procs are when they GET attacked, rather than when THEY attack. This is so confusing and messed up, it has demotivated me so much that I think I'm calling it quits now.

**Comment by pls003** [comment]
You didn't know that?

**Comment by IndieanaJones** [comment]
You didn't know that?

**Pull request closed without merging**